src/ImageProcessorCore/Colors/ColorTransforms.cs

```diff
@@ -18,9 +19,60 @@ namespace ImageProcessorCore
     public partial struct Color
     {
         /// <summary>
+        /// Blends two colors by lightening.
+        /// </summary>
+        /// <remarks>
+        /// Selects the lighter of the backdrop and source colors.
+        /// The backdrop is replaced with the source where the backdrop is lighter; otherwise, it is left unchanged
+        /// Note: This behavior is on a channel by channel basis, i.e., this rule is applied to each of the 3 RGB color channels separately.
+        /// </remarks>
+        /// <param name="backdrop"></param>
+        /// <param name="source"></param>
+        /// <returns>
+        /// The <see cref="Color"/>.
+        /// </returns>
+        public static Color Lighten(Color backdrop, Color source)
+        {
+            var newVector = new Vector4
```

Comment:
This is from the w3c spec:

The result of the mixing function is modulated by the backdrop alpha. A 
fully opaque backdrop allows the mixing function to be fully realized. A 
transparent backdrop will cause the final result to be a weighted average 
between the source color and mixed color with the weight controlled by the 
backdrop alpha. The value of the new color becomes:

Cr = (1 - αb) x Cs + αb x B(Cb, Cs)

with:

Cr: the result color
B: the formula that does the blending
Cs: the source color
Cb: the backdrop color
αb: the backdrop alpha

-----------------------------

Which is why I've been hemming and hawing about pre multiplied alphas. If 
we have already added the alpha as part of the color in the library than we 
would need to undo that to fit the spec because the blending function is 
modulated by the alpha before the blend is applied to the color.


On February 15, 2016 12:00:26 AM James Jackson-South 
<notifications@github.com> wrote:

>> @@ -18,9 +19,60 @@ namespace ImageProcessorCore
>>      public partial struct Color
>>      {
>>          /// <summary>
>> +        /// Blends two colors by lightening.
>> +        /// </summary>
>> +        /// <remarks>
>> +        /// Selects the lighter of the backdrop and source colors.
>> +        /// The backdrop is replaced with the source where the backdrop is 
>> lighter; otherwise, it is left unchanged
>> +        /// Note: This behavior is on a channel by channel basis, i.e., 
>> this rule is applied to each of the 3 RGB color channels separately.
>> +        /// </remarks>
>> +        /// <param name="backdrop"></param>
>> +        /// <param name="source"></param>
>> +        /// <returns>
>> +        /// The <see cref="Color"/>.
>> +        /// </returns>
>> +        public static Color Lighten(Color backdrop, Color source)
>> +        {
>> +            var newVector = new Vector4
>
> Since they are static methods on the `Color` struct they would be used in 
> either scenario. Looking at the spec I would say it should take the higher 
> of the two alpha values.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/JimBobSquarePants/ImageProcessor/pull/329/files#r52859768